jvm/core/src/main/scala/org/scalatest/tools/Framework.scala

```diff
@@ -886,8 +885,23 @@ import java.net.{ServerSocket, InetAddress}
               case e: RunCompleted => // Sub-process completed, just let the thread terminate
               case e: RunStopped => dispatchReporter(e)
               case e: RunAborted => dispatchReporter(e)
-	        }
+            }  
           }
+
+          @annotation.tailrec
+          final def tryReact(): Unit = 
+            try {
+              react()  
+            }
+            catch {
+              case t: Throwable => 
+                // Restart server socket
```

Comment:
@cheeseng Here's another catch of Throwable. Should this be NotSerializableException?